tests/lib/rules/no-empty-function.js

```diff
@@ -386,3 +390,317 @@ ruleTester.run(
+			allow: "getters",
+		},
+		{
+			code: "var A = class {static get foo(): string {}};",
+			messageId: "unexpected",
+			data: { name: "static getter 'foo'" },
+			allow: "getters",
+		},
+		{
+			code: "var obj = {set foo(value: string) {}};",
+			messageId: "unexpected",
+			data: { name: "setter 'foo'" },
+			allow: "setters",
+		},
+		{
+			code: "class A {set foo(value: string) {}}",
+			messageId: "unexpected",
+			data: { name: "setter 'foo'" },
+			allow: "setters",
+		},
+		{
+			code: "class A {static set foo(value: string) {}}",
+			messageId: "unexpected",
+			data: { name: "static setter 'foo'" },
+			allow: "setters",
+		},
+		{
+			code: "var A = class {set foo(value: string) {}};",
+			messageId: "unexpected",
+			data: { name: "setter 'foo'" },
+			allow: "setters",
+		},
+		{
+			code: "var A = class {static set foo(value: string) {}};",
+			messageId: "unexpected",
+			data: { name: "static setter 'foo'" },
+			allow: "setters",
+		},
+		{
+			code: "class A { constructor(param: string) {} }",
+			messageId: "unexpected",
+			data: { name: "constructor" },
+			allow: "constructors",
+		},
+		{
+			code: "var A = class {constructor(param: string) {}};",
+			messageId: "unexpected",
+			data: { name: "constructor" },
+			allow: "constructors",
+		},
+		{
+			code: "const foo = { async method(param: string) {} }",
+			allow: "asyncMethods",
+			messageId: "unexpected",
+			data: { name: "async method 'method'" },
+		},
+		{
+			code: "async function a(param: string){}",
+			allow: "asyncFunctions",
+			messageId: "unexpected",
+			data: { name: "async function 'a'" },
+		},
+		{
+			code: "const foo = async function(param: string) {}",
+			messageId: "unexpected",
+			data: { name: "async function" },
+			allow: "asyncFunctions",
+		},
+		{
+			code: "class A { async foo(param: string) {} }",
+			messageId: "unexpected",
+			data: { name: "async method 'foo'" },
+			allow: "asyncMethods",
+		},
+		{
+			code: "const foo = async (): Promise<void> => {};",
+			messageId: "unexpected",
+			data: { name: "async arrow function" },
+			allow: "arrowFunctions",
+		},
+		{
+			code: "class A { private constructor() {} }",
+			messageId: "unexpected",
+			data: { name: "constructor" },
+			allow: "privateConstructors",
+		},
+		{
+			code: "var A = class { private constructor() {} };",
+			messageId: "unexpected",
+			data: { name: "constructor" },
+			allow: "privateConstructors",
+		},
+		{
+			code: "class A { protected constructor() {} }",
+			messageId: "unexpected",
+			data: { name: "constructor" },
+			allow: "protectedConstructors",
+		},
+		{
+			code: "var A = class { protected constructor() {} };",
+			messageId: "unexpected",
+			data: { name: "constructor" },
+			allow: "protectedConstructors",
+		},
+		{
+			code: "class A { @decorator() foo() {} }",
+			messageId: "unexpected",
+			data: { name: "method 'foo'" },
+			allow: "decoratedFunctions",
+		},
+		{
+			code: "var A = class { @decorator() foo() {} }",
```

Comment:
We are preferring const in latest examples & test cases.
```suggestion
			code: "const A = class { @decorator() foo() {} }",
```

Reply:
js ones are fine as is, I meant the ts ones that we are adding newly.